### DISCUSSION TITLE
GraphQL query returns unpredictable results

### DISCUSSION THREAD

**Discussion by lampersky** [OrchardCMS/OrchardCore#9237] [Q&A]
Is there any known reason why particular GraphQL query behave unpredictable?
Once in a while I'm getting error like this:
![image](https://user-images.githubusercontent.com/6403130/115392525-f3cd5b00-a1e0-11eb-8da4-305600abdbae.png)

Usually my query works as it should, 1 out of 20 requests and I'm getting this error.

My GraphQL query looks similar to this, maybe it's a bit more complex, more fields etc.
```
query MyQuery {
  someContentType {
    list {
      contentItems {
        ... on Material {
          displayText
          bag {
            contentItems {
              ... on MaterialItem {
                caption
                description
                url
              }
            }
          }
        }
      }
    }
  }
}
```
But I don't think there is a problem with query itself, because 95% of requests return proper results.
Has anyone spotted similar problem?

**Comment by lampersky** [comment]
https://user-images.githubusercontent.com/6403130/115730762-ddf99a80-a386-11eb-9a0c-dccacc859e80.mp4

**Reply by lampersky** [reply]
@jtkech maybe there is some race condition while resolving content items? I can't find any explanation for this problem.
Your thoughts would be helpful.

**Reply by jtkech** [reply]
@lampersky 

Sorry i'm quite busy on other things ;)

Just tried differents queries with the last dev, i can't repro

Which OC version are you using?

Can you share a sample app with a sample query so that i could repro?

**Reply by lampersky** [reply]
@jtkech  I'm using latest dev, but I was able to reproduce this with oldest version (before graphql 4 dependency).
I will try to prepare repro repo, but it is not easy because of sensitive data.
